src/MongoDB.Driver.Core/Core/Authentication/Sspi/SspiSecurityContext.cs

```diff
@@ -334,6 +334,17 @@ protected override bool ReleaseHandle()
             return NativeMethods.DeleteSecurityContext(ref _sspiHandle) == 0;
         }
 
+        protected override void Dispose(bool disposing)
+        {
+            if (!_isDisposed && disposing)
+            {
+                _credential?.Dispose();
```

Comment:
Was it a bug that this wasn't already being disposed?

Reply:
Prior to the refactoring, `SecurityCredential` wasn't owned by `SspiSecurityContext`. Instead both were registered separately for disposal with `SaslConversation`. Now `SspiSecurityContext` owns the `SecurityContext` and is responsible for its cleanup. This also makes a lot more sense for GSSAPI since we allocate many unmanaged buffers/objects for interacting with the GSSAPI library and having one place to clean everything up will make the code easier to understand.

Reply:
Thanks, makes sense.

Reply:
We still register `credential` for disposing here: https://github.com/JamesKovacs/mongo-csharp-driver/pull/4/files?file-filters%5B%5D=.cs#diff-ed247e1079c0fac7539516f899044f39a5d92d9b0ad8703fd6f23ea95240268cR18 I would vote to do such logic in one place. Also, if I'm not mistaken, if we move disposing for _credentials to this line, the whole `conversation.RegisterItemForDisposal` became unused, doesn't it?